graphql_compiler/compiler/emit_sql.py

```diff
@@ -876,25 +902,29 @@ def unfold(self):
         self._aliases[subquery_alias_key] = fold_subquery
         self._from_clause = from_cls
 
-        # Ensure references to the outer vertex table after the Unfold refer to a totally new
-        # copy of the outer vertex table. Otherwise references would select columns from the
-        # copy of that table found inside the fold subquery.
-        self._aliases[(self._current_location.at_vertex().query_path, None)] = outer_vertex
-
         # clear the fold from the compilation state
         self._current_fold = None
         self._fold_vertex_location = None
         self._current_alias = outer_vertex
+        self._current_location = self._current_location.base_location
 
-    def mark_location(self):
+    def mark_location(self, is_last_fold_block):
```

Comment:
The meaning of `is_last_fold_block` is not obvious at all. Either document it well or compute it inside this function (passing the whole ir_block in as an argument). The fact that the very definition of this concept depends on the restriction that there's only one output scope inside fold makes me think that we should not glorify it as a named concept leaking outside the `CompilationState` class.

Reply:
Good idea, done.